nextflow_schema.json

```diff
@@ -0,0 +1,3239 @@
ent ion. By doing that, the number of searches needed for identification will significantly decrease. The rest of the ions that were matched against features without annotation will be searched using different adduct rules and charges. The resulting scores from metabolite spectrum matches will be transformed into posterior error probability and re-matched to the features at the later step.\n\nCurrently, MetaboIGNITER supports two types of identification. One is based on in-silico (database) approach and the other is based on internal library.\n\nIf you would like to perform identification, please set the following parameters to true:\n\n```nextflow\n    perform_identification=true\n```\n\nPlease also remember that we only perform identification for the ionization modes that you have quantification data for.\n\n## In-silico identification in MetaboIGNITER\n\nCurrently, MetaboIGNITER supports three search engines, [MetFrag](https://ipb-halle.github.io/MetFrag/), [CSI:FingerID](https://bio.informatik.uni-jena.de/software/sirius/), and [CFM-ID](https://cfmid.wishartlab.com/) for performing identification. These engines share some global parameters but also each of these will need specific set of parameters. The user can select multiple search engines to do the identification. In the case of multiple search engines, the workflow will have multiple final output, one for each search engine and ionization. Please set the parameter to true for the desired search engine:\n\n```nextflow\n    perform_identification_metfrag=false \n    perform_identification_csifingerid=false\n    perform_identification_cfmid=false \n```\n\nCharacterize your own library\n\nThis part of the workflow is used to create and characterize in-house library. This is how it works: we assume that the library consists of one or more mzML files, each containing a number of compounds. A possible scenario is when the users have several standard metabolites that can have overlapping masses with unknown retention times. The standards with overlapping masses can be run separately using MS, resulting in different runs. MetaboIGNITER will help you to characterize this type of internal libraries. You will need to construct the Characterization file (see below) that shows which standards are present in which mzML file. The workflow will then do mass trace detection, MS2 extraction and mapping of parent ions to mass traces. Doing so will result in finding the retention time and empirical m/z of each standard. This will then be used to create identification parameters and search the biological MS2 files.\n\nSet the library parameter to true if you would like to perform library search:\n\n```nextflow\n    perform_identification_internal_library=true\n```\n\n## Output files\nThere will be a lot of different files generated by the workflow, if you are interested only in the final output part and identification information, please set this flag to false\n\n```nextflow\n    publishDir_intermediate\n```"
```

Comment:
```suggestion
            "help_text": "Centroiding\n\nWe recommend inputting already centroided files. You can achieve this at the conversion steps in msconvert. However, if your data is not centroided, you can let the workflow doing that for you. We use OpenMS \"PeakPickHiRes\" tool to perform that. Set the following parameter to *true* to perform the centroiding:\n\n```nextflow\n    need_centroiding\n```\n\nPlease be aware that setting need_centroiding=true will do centroiding on all of your data including both ionization, identification etc.\n\nIf you want non-defualt values, to control the parameters centroiding you can edit *openms_peak_picker_ini_pos.ini* and *openms_peak_picker_ini_neg.ini* files located under *metaboigniter/assets/openms*. The description of the parameters can be found on [OpenMS website](https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/nightly/html/TOPP_PeakPickerHiRes.html).\n\nIf you choose to perform the quantification using OpenMS, you should consider changing the parameters for OpenMS only. You can edit *openms_feature_finder_metabo_ini_pos.ini* and *openms_feature_finder_metabo_ini_neg.ini* files located under *metaboigniter/assets/openms*. The description of the parameters can be found on [OpenMS website](https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/nightly/html/TOPP_FeatureFinderMetabo.html). **When tuning the OpenMS parameters make sure that \"report_convex_hulls\" is set to true**\n\n## Ionization\n\nSet what type of ionization you have. You can either set to 'pos' (only positive), 'neg' (only negative), 'both' (both positive and negative):\n\n```nextflow\n      type_of_ionization\n```\n\n## Identification Specific Parameters\n\nMetaboIGNITER has a specific design for performing identification. After performing quantification and annotation of MS1 data, the results will be fed into the identification sub-pipeline this part of the workflow, will first extract the MS2 information from the identification mzML files and map them to the quantification data. If a parent ion was matched against a feature that was successfully annotated, we then estimate the neutral mass for that parent ion. By doing that, the number of searches needed for identification will significantly decrease. The rest of the ions that were matched against features without annotation will be searched using different adduct rules and charges. The resulting scores from metabolite spectrum matches will be transformed into posterior error probability and re-matched to the features at the later step.\n\nCurrently, MetaboIGNITER supports two types of identification. One is based on in-silico (database) approach and the other is based on internal library.\n\nIf you would like to perform identification, please set the following parameters to true:\n\n```nextflow\n    perform_identification=true\n```\n\nPlease also remember that we only perform identification for the ionization modes that you have quantification data for.\n\n## In-silico identification in MetaboIGNITER\n\nCurrently, MetaboIGNITER supports three search engines, [MetFrag](https://ipb-halle.github.io/MetFrag/), [CSI:FingerID](https://bio.informatik.uni-jena.de/software/sirius/), and [CFM-ID](https://cfmid.wishartlab.com/) for performing identification. These engines share some global parameters but also each of these will need specific set of parameters. The user can select multiple search engines to do the identification. In the case of multiple search engines, the workflow will have multiple final output, one for each search engine and ionization. Please set the parameter to true for the desired search engine:\n\n```nextflow\n    perform_identification_metfrag=false \n    perform_identification_csifingerid=false\n    perform_identification_cfmid=false \n```\n\nCharacterize your own library\n\nThis part of the workflow is used to create and characterize in-house library. This is how it works: we assume that the library consists of one or more mzML files, each containing a number of compounds. A possible scenario is when the users have several standard metabolites that can have overlapping masses with unknown retention times. The standards with overlapping masses can be run separately using MS, resulting in different runs. MetaboIGNITER will help you to characterize this type of internal libraries. You will need to construct the Characterization file (see below) that shows which standards are present in which mzML file. The workflow will then do mass trace detection, MS2 extraction and mapping of parent ions to mass traces. Doing so will result in finding the retention time and empirical m/z of each standard. This will then be used to create identification parameters and search the biological MS2 files.\n\nSet the library parameter to true if you would like to perform library search:\n\n```nextflow\n    perform_identification_internal_library=true\n```\n\n## Output files\nThere will be a lot of different files generated by the workflow, if you are interested only in the final output part and identification information, please set this flag to false\n\n```nextflow\n    publishDir_intermediate\n```"
```

Reply:
Aaah. Then i completely misunderstood  it! Will fix. More comments?

Reply:
The problem appears to be fixed. Thanks for the suggestion!